File: C#/백준/Silver/10826. 피보나치 수 4/피보나치 수 4.cs
using System;
using System.Numerics;

namespace q{
    class w{
        static void Main(){
            int a = int.Parse(Console.ReadLine());
            if (a<=1){
                Console.WriteLine(a);
            }
            else{
                BigInteger n1 = 0;
                BigInteger n2 = 1;
                BigInteger ans = 0;
                for (int i=0;i<a-1;i++){
                    ans = n1+n2;
                    n1 = n2;
                    n2 = ans;
                }
                Console.WriteLine(ans);
            }
        }
    }
}

File: Java/백준/Silver/10826. 피보나치 수 4/피보나치 수 4.java
import java.util.Scanner;
import java.math.BigInteger;

public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int a = in.nextInt();
        if(a<=1){
            System.out.println(a);
        }
        else{
            BigInteger n1 = new BigInteger("0");
            BigInteger n2 = new BigInteger("1");
            for (int i=0;i<a-1;i++){
                BigInteger ans =  n1.add(n2);
                n1 = n2;
                n2 = ans;
            }
            System.out.println(n2);
        }
    }
}